app/src/org/commcare/utils/FileUtil.java

```diff
@@ -857,4 +866,79 @@ public static boolean isFileTooLargeToUpload(ContentResolver contentResolver, Ur
             return returnCursor.getLong(sizeIndex) > FormUploadUtil.MAX_BYTES;
         }
     }
+
+    public static void copyFileWithExifData(File sourceFile, File destFile) throws IOException {
+        // First copy the file normally
+        copyFile(sourceFile, destFile);
+        
+        // Only try to copy EXIF data for image files
+        String mimeType = getMimeType(sourceFile.getAbsolutePath());
+        if (mimeType != null && mimeType.startsWith("image/")) {
+            try {
+                copyExifData(sourceFile.getAbsolutePath(), destFile.getAbsolutePath());
+            } catch (Exception e) {
+                Logger.log(LogTypes.TYPE_WARNING_NETWORK,
+                    "Failed to copy EXIF data: " + e.getMessage());
+            }

```

Comment:
Seems like we are logging exceptions at 2 places with `copyExifData` being the other. Is it possible to just do it inside `copyExifData` and make it fail safe.